lib/npg_qc/autoqc/checks/rna_seqc.pm

```diff
@@ -0,0 +1,328 @@
+package npg_qc::autoqc::checks::rna_seqc;
+
+use Moose;
+use namespace::autoclean;
+use English qw( -no_match_vars );
+use Carp;
+use File::Spec::Functions qw( catdir );
+use File::Basename;
+use File::Path qw(make_path);
+use npg_qc::autoqc::types;
+use npg::api::run;
+use Readonly;
+use Try::Tiny;
+use DateTime;
+use npg_tracking::util::types;
+
+extends qw(npg_qc::autoqc::checks::check);
+
+with qw(npg_tracking::data::reference::find 
+        npg_common::roles::software_location
+        npg_tracking::data::transcriptome::find
+       );
+
+our $VERSION = '0';
+
+Readonly::Scalar our $EXT => q[bam];
+Readonly::Scalar my $RNASEQC_JAR_NAME       => q[RNA-SeQC_v1.1.8.jar];
+Readonly::Scalar my $RNASEQC_JAR_VERSION    => q[1.1.8];
+Readonly::Scalar my $RNASEQC_GTF_TTYPE_COL  => 2;
+Readonly::Scalar my $RNASEQC_GTF_DIRECTORY  => q[RNA-SeQC];
+Readonly::Scalar my $RNASEQC_OUTDIR         => q[rna_seqc];
+Readonly::Scalar my $JAVA_MAX_HEAP_SIZE     => q[4000m];
+Readonly::Scalar my $JAVA_GC_TYPE           => q[+UseSerialGC];
+Readonly::Scalar my $JAVA_USE_PERF_DATA     => q[-UsePerfData];
+
+has '+file_type' => (default => $EXT,);
+
+has '+aligner' => (default => q[fasta],);
+
+has 'java_max_heap_size'     => (is      => 'ro',
+                                 isa     => 'Str',
+                                 default => $JAVA_MAX_HEAP_SIZE,);
+
+has 'java_gc_type'           => (is      => 'ro',
+                                 isa     => 'Str',
+                                 default => $JAVA_GC_TYPE,);
+
+has 'java_use_perf_data'     => (is      => 'ro',
+                                 isa     => 'Str',
+                                 default => $JAVA_USE_PERF_DATA,);
+
+has 'java_jar_path'          => (is      => 'ro',
+                                 isa     => 'NpgCommonResolvedPathJarFile',
+                                 coerce  => 1,
+                                 default => $RNASEQC_JAR_NAME,);
+
+has 'transcript_type' => (is      => 'ro',
+                          isa     => 'Int',
+                          default => $RNASEQC_GTF_TTYPE_COL,);
+
+
+has 'alignments_in_bam' => (is         => 'ro',
+                            isa        => 'Maybe[Bool]',
+                            lazy_build => 1,);
+
+sub _build_alignments_in_bam {
+    my ($self) = @_;
+    return $self->lims->alignments_in_bam;
+}
+
+###############################################################################
+# Output path:
```

Comment:
The pipeline-related explanations should be in the pipeline. Here they are somewhat out of place.